### DISCUSSION TITLE
Ghub works but forge doesn't for self-hosted gitlab instance

### DISCUSSION THREAD

**Discussion by theottm** [magit/forge#590] [Support] [answered]
Hello,

When running `f n` in Magit status buffer I get
```quote
Cannot determine forge repository.  git@gitlab.myinstance.xyz:myproject/myproject.git isn’t a forge url
``` 

Here are the relevant local git variables:
```quote
gitlab.myinstance.xyz/api/v4.user=teddd
gitlab.host=myinstance.xyz/api/v4
remote.origin.url=git@gitlab.myinstance.xyz:myproject/myproject.git
```
I have no global git variables relative to any forge.

Despite `forge` not working, I am able to interact with the api of gitlab with `ghub` using 
```elisp
(ghub-request "GET" "/user" nil
              :forge 'gitlab
              :host "gitlab.myinstance.xyz/api/v4"
              :username "teddd"
              :auth 'ghub)
```
I have a line for both users `teddd^forge` and `teddd^ghub` in `.authinfo`.

It seems like forge is not able to use the information of git config. Did I miss something? I also tried to add git variable `forge.remote=myinstance.xyz/api/v4` with no success.

Cheers,

Théo

**Comment by tarsius** [comment, marked as answer]
Sounds like you didn't add the host to `forge-alist`.

**Reply by theottm** [reply]
Worked, thanks.

**Reply by zzantares** [reply]
is this in the manual? I found about `forge-alist` after searching around in gh issues; also I wonder, there's `forge-add-repository` any reason why we couldn't store forges as well and have `forge-add` so we don't have to add it to `forge-alist` manually?

Thanks.

**Reply by tarsius** [reply]
It was mentioned in the relevant places, but I have now copied the docstring into the manual and added an additional cross-reference (adce8470e7f57af4091eac294f5894055084ca40), to make it less likely that it is overlooked.
